### PR TITLE
[SPARK-5974] [SPARK-5980] [mllib] [python] [docs] Update ML guide with save/load, Python GBT

### PR DESCRIPTION
**Pull request by jkbradley**
- Add GradientBoostedTrees Python examples to ML guide
  - I ran these in the pyspark shell, and they worked.
- Add save/load to examples in ML guide
- Added note to python docs about predict,transform not working within RDD actions,transformations in some cases (See SPARK-5981)

CC: @mengxr

### CODE REVIEW

File: docs/mllib-collaborative-filtering.md [hunk 1]
```diff
@@ -95,6 +95,9 @@ val MSE = ratesAndPreds.map { case ((user, product), (r1, r2)) =>
   err * err
 }.mean()
 println("Mean Squared Error = " + MSE)
+
+model.save("myModelPath")
+val sameModel = MatrixFactorizationModel.load("myModelPath")
```
Comment (mengxr) [inline docs/mllib-collaborative-filtering.md]: import `MatrixFactorizationModel` and please check others

File: python/pyspark/mllib/feature.py [hunk 2]
```diff
@@ -120,6 +120,9 @@ def transform(self, vector):
         """
         Applies standardization transformation on a vector.
 
+        Note: In Python, transform cannot currently be used within an RDD
+              transformation or action.  Call transform directly on the RDD instead.
```
Comment (mengxr) [inline python/pyspark/mllib/feature.py]: This line may be too long for Python doc. We use 72 in PySpark, the same as in PEP8.

**Comment by jkbradley** [comment]
OK that should fix those 2 issues

**Comment by jkbradley** [comment]
Note I corrected a bunch of other doc lines in those files too but did not edit the text content at all.

**Comment by mengxr** [comment]
LGTM. Merged into master and branch-1.3. Thanks!

**Pull request closed without merging**